bokehjs/src/lib/models/glyphs/hstrip.ts

```diff
@@ -0,0 +1,214 @@
+import {Glyph, GlyphView, GlyphData} from "./glyph"
+import {Selection} from "../selections/selection"
+import {LineVector, FillVector, HatchVector} from "core/property_mixins"
+import {PointGeometry, SpanGeometry, RectGeometry} from "core/geometry"
+import {FloatArray, ScreenArray, Rect} from "core/types"
+import * as visuals from "core/visuals"
+import * as uniforms from "core/uniforms"
+import {Context2d} from "core/util/canvas"
+import {SpatialIndex} from "core/util/spatial"
+import {map} from "core/util/arrayable"
+import {range} from "core/util/array"
+import * as p from "core/properties"
+
+export type HStripData = GlyphData & p.UniformsOf<HStrip.Mixins> & {
+  _y0: FloatArray
+  _y1: FloatArray
+
+  sy0: ScreenArray
+  sy1: ScreenArray
+
+  max_line_width: number
+}
+
+export interface HStripView extends HStripData {}
+
+export class HStripView extends GlyphView {
+  declare model: HStrip
+  declare visuals: HStrip.Visuals
+
+  /** @internal */
+  declare glglyph?: import("./webgl/lrtb").LRTBGL
+
+  override async lazy_initialize(): Promise<void> {
+    await super.lazy_initialize()
+
+    const {webgl} = this.renderer.plot_view.canvas_view
+    if (webgl != null && webgl.regl_wrapper.has_webgl) {
+      const {LRTBGL} = await import("./webgl/lrtb")
+      this.glglyph = new LRTBGL(webgl.regl_wrapper, this)
+    }
+  }
+
+  get sleft(): ScreenArray {
+    const {left} = this.renderer.plot_view.frame.bbox
+    const n = this.data_size
+    const sleft = new ScreenArray(n)
+    sleft.fill(left)
+    return sleft
+  }
+
+  get sright(): ScreenArray {
+    const {right} = this.renderer.plot_view.frame.bbox
+    const n = this.data_size
+    const sright = new ScreenArray(n)
+    sright.fill(right)
+    return sright
+  }
+
+  get stop(): ScreenArray {
+    return this.sy0
+  }
+
+  get sbottom(): ScreenArray {
+    return this.sy1
+  }
+
+  override after_visuals(): void {
+    super.after_visuals()
+    this.max_line_width = uniforms.max(this.line_width)
+  }
+
+  protected override _index_data(index: SpatialIndex): void {
+    const {_y0, _y1, data_size} = this
+
+    for (let i = 0; i < data_size; i++) {
+      const y0_i = _y0[i]
+      const y1_i = _y1[i]
+      index.add_rect(0, y0_i, 0, y1_i)
```

Comment:
This verges on clever. It might be worth a comment to explain how this will be used and why it's reasonable to just choose a convenient arbitrary x-location